sky/services/platform/ios/path_provider_impl.mm

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/services/platform/ios/path_provider_impl.h"
+#include "base/mac/scoped_nsautorelease_pool.h"
+
+#include <Foundation/Foundation.h>
+
+#include <string>
+
+namespace flutter {
+namespace platform {
+
+PathProviderImpl::PathProviderImpl(mojo::InterfaceRequest<PathProvider> request)
+    : binding_(this, request.Pass()) {}
+
+PathProviderImpl::~PathProviderImpl() {}
+
+void PathProviderFactory::Create(mojo::ApplicationConnection* connection,
+                                 mojo::InterfaceRequest<PathProvider> request) {
+  new PathProviderImpl(request.Pass());
+}
+
+static std::string GetDirectoryOfType(NSSearchPathDirectory dir) {
+  base::mac::ScopedNSAutoreleasePool pool;
```

Comment:
What's the significance of the autorelease pool in these implementations?  If they're always required, I wonder if they could be hoisted into a base class or into the service infrastructure.